### ISSUE TITLE
Issues running examples with empty Environment and attempting to compute atmospheric loss

### ISSUE THREAD

**Issue by inaki-ortizdelandaluce** [Free-Space-QKD-Lab-HWU/Qrackling#50]
I am experiencing some issues when running certain scripts under the Examples folder. I am adding to the path the Examples and the uplink_adaptive_optics folders and subfolders and then, whenever I try to run ExampleOrbitBB84 or some others (e.g.  ExampleOrbitCOW, ExampleOrbitDPS) I get some errors related to interpolation as below:

````
>> ExampleOrbitBB84

correct_size =

     1     1     1

Error using [griddedInterpolant](matlab:matlab.lang.internal.introspective.errorDocCallback('griddedInterpolant'))
Interpolation requires at least two sample points for each grid dimension.

Error in [interpn](matlab:matlab.lang.internal.introspective.errorDocCallback('interpn', '/MATLAB/toolbox/matlab/polyfun/interpn.m', 149)) ([line 149](matlab: opentoline('/MATLAB/toolbox/matlab/polyfun/interpn.m',149,0)))
           F = griddedInterpolant(X, V, method,extrap);

Error in [environment.Environment/Interp](matlab:matlab.lang.internal.introspective.errorDocCallback('environment.Environment/Interp', '/MATLAB Drive/Qrackling20240520/+environment/@Environment/Environment.m', 185)) ([line 185](matlab: opentoline('/MATLAB Drive/Qrackling20240520/+environment/@Environment/Environment.m',185,0)))
                interp_data = interpn( ...

Error in [nodes.AtmosphericLoss](matlab:matlab.lang.internal.introspective.errorDocCallback('nodes.AtmosphericLoss', '/MATLAB Drive/Qrackling20240520/+nodes/AtmosphericLoss.m', 32)) ([line 32](matlab: opentoline('/MATLAB Drive/Qrackling20240520/+nodes/AtmosphericLoss.m',32,0)))
    loss = environment.Interp("attenuation", abs(headings), abs(elevations), wavelength);

Error in [nodes.linkLoss](matlab:matlab.lang.internal.introspective.errorDocCallback('nodes.linkLoss', '/MATLAB Drive/Qrackling20240520/+nodes/linkLoss.m', 71)) ([line 71](matlab: opentoline('/MATLAB Drive/Qrackling20240520/+nodes/linkLoss.m',71,0)))
                res = nodes.AtmosphericLoss(kind, receiver, transmitter, options.environment);

Error in [nodes.QkdPassSimulation](matlab:matlab.lang.internal.introspective.errorDocCallback('nodes.QkdPassSimulation', '/MATLAB Drive/Qrackling20240520/+nodes/QkdPassSimulation.m', 67)) ([line 67](matlab: opentoline('/MATLAB Drive/Qrackling20240520/+nodes/QkdPassSimulation.m',67,0)))
        [link_loss, ~] = nodes.linkLoss("qkd", Receiver, Transmitter, ...

Error in [ExampleOrbitBB84](matlab:matlab.lang.internal.introspective.errorDocCallback('ExampleOrbitBB84', '/MATLAB Drive/Qrackling20240520/Examples/ExampleOrbitBB84.m', 49)) ([line 49](matlab: opentoline('/MATLAB Drive/Qrackling20240520/Examples/ExampleOrbitBB84.m',49,0)))
Result=nodes.QkdPassSimulation(SimGround_Station,SimSatellite,protocol.bb84);
````
The problem seems to be related with attempting to interpolate with an empty Environment, but I am not fully sure about that. Check in QkdPassSimulation.m:Line 66 may be problematic:

````
    if ~isempty(options.Environment)
        [link_loss, ~] = nodes.linkLoss("qkd", Receiver, Transmitter, ...
            "apt", "optical", "geometric", "turbulence", ...
            "atmospheric", "environment", options.Environment);
````
Replacing the emptiness check by this one, seems to fix the issue when running again the failing scripts under the Examples folder:
````
if ~isempty(options.Environment) & ~isequal(options.Environment, environment.Environment.empty)
````

I take the chance to share my deepest gratitude to make this toolbox available.

**Comment by Peter-Barrow** [comment]
Hi, thank you for filing this and pointing us in the direction of the suspicious code it's much appreciated!

This has been corrected on main now (fix in https://github.com/Free-Space-QKD-Lab-HWU/Qrackling/commit/cd0db3fbdd9b1c3bf30ff4458cd43fd4c5c96c7c). QkdPassSimulation now takes an Environment as an optional argument instead of supplying an empty Environment as a place holder.

> I take the chance to share my deepest gratitude to make this toolbox available.

We greatly appreciate this sentiment and are glad to see people using Qrackling, please don't hesitate to get in contact with our team or post further issues!
